src/main/java/org/opentripplanner/middleware/models/TrackedJourney.java

```diff
@@ -45,8 +46,14 @@ public TrackedJourney(StartTrackingPayload trackingPayload) {
         tripId = trackingPayload.tripId;
     }
 
+    /**
+     * Remove duplicates before adding to tracked locations.
+     */
     public void update(UpdatedTrackingPayload trackingPayload) {
-        this.locations.addAll(trackingPayload.locations);
+        LinkedHashSet<TrackingLocation> trackedLocations = new LinkedHashSet<>(locations);
```

Comment:
Let's just revert that for now. I think there is more to that. My previous suggestion was more about handling if the same update request is sent twice but that's more a nice-to-have.